enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
+
+2. What to do(if anything) for components with interdependencies, to ensure a user doesn't break
+enabled components by disabling a dependency?  Options include:
+
+* Do nothing other than document dependencies so users know what not to turn off
+* Don't even annotate dependency resources for filtering, so if something is a dependency it cannot be turned off
+* Logic in the install or CVO that intelligently analyzes the filters the user has supplied and checks
+  for dependency issues (least desirable solution imho).
+
+3. Is Capabilities an ok name for these things/fields?  We'll want to get the naming right on the api to
+convey the right expectations for users.  We may revisit this name during implementation.
+
+### Test Plan
+
+1) Install clusters w/ the various add-on components included/excluded and confirm
+that the cluster is functional but only running the expected add-ons.
+
+2) Upgrade a cluster to a new version that includes new resources that belong to
+an addon that was included in the original install.  The new resources should be
+created.
+
+3) Upgrade a cluster to a new version that includes new resources that belong to
+an addon that was excluded in the original install.  The new resources should *not* be
+created.
+
+4) After installing a cluster, enable additional addons.  The newly enabled addons should
+be installed/reconciled by the CVO.
+
+5) After installing a cluster, disable an addon.  The configuration change should be
+rejected by the CVO.  Disabling a component post-install is not supported.
+
+
+
+### Graduation Criteria
+
+Would expect this to go directly to GA once a design is agreed upon/approved.
+
+#### Dev Preview -> Tech Preview
+N/A
+
+#### Tech Preview -> GA
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+For upgrades, any new resources should have the same exclusion filters applied by the CVO.
+For downgrades, if downgrading below the version of the CVO that supports this logic
+previously excluded components will get created on the cluster.  This is likely
+not a concern since you can't downgrade below the version you started at, and if
+you're using this feature that means you started at a version of the CVO that supports it.
+
+If we allow enabling filters post-install, then we need to revisit the implications of
+downgrades.
+
+There is also some risk if a particular resource has different annotations in different
+versions, then upgrading/downgrading could change whether that resource is excluded by
+the CVO or not.  Once created, the CVO never deletes resources, so some manual cleanup
+might be needed to achieve the desired state.  For downgrades this is probably acceptable,
```

Comment:
#414 is how the CVO can be told to delete resources, and that's live today.  So you can probably cover most transitions by grouping the right deletion manifests into the component-removed situation.  With all of the usual caveats that sometimes [generic uninstall logic is difficult][1], and you may need some component-specific special handling.

[1]: https://github.com/openshift/enhancements/pull/922#discussion_r731190381